aten/src/ATen/native/cuda/Lerp.cu

```diff
@@ -52,7 +52,7 @@ Tensor& lerp_cuda_tensor_out(Tensor& result, const Tensor& self,
   TORCH_CHECK(weight.dim() <= std::max(self.dim(), end.dim()),
            "weight should be of dimension max(self.dim(), end.dim()) or lesser");
   std::tie(b_self, b_end, b_weight) = expand_outplace(self, end, weight, "lerp_out_cuda");
-  result.resize_as_(b_self, b_self.suggest_memory_format());
+  result.resize_as_(b_self);
```

Comment:
Actually, can you try not resizing result here (and in other places) at all, and letting tensorIterator figure it out? That way, correct memory format should be inferred automatically.

Reply:
I removed them and I also added back suggest_memory_format() when out is not given since we create the result tensor in that case.

Reply:
When out is not given, either of 2 things can be done:
1) undefined output sent to TensorIterator
2) zero-sized output sent to TensorIterator. 
First is preferred when there's also type promotion to consider, so the caller doesn't know which scalar type to use for zero-sized tensor. Second can be used when there's no type promotion and/or the type of the output is known.
But the caller should not be setting out memory format.

Reply:
TODO:

There are still some Lerp cases calling empty_like where they shouldn't. For example:

```
Tensor result = at::empty_like(b_self, b_self.suggest_memory_format());
```

on line 100. It'd be good to clean those up, too, for consistency. 

@heitorschueroff (after nanmedian, if you're looking for a small PR to pickup).